API and Gateway v8 (#2097)
hammerandchisel/discord-api-docs: docs/topics/Gateway.md

###### Gateway Versions

| Version | Status       |
|---------|--------------|
| 8       | Available    |

> | 6       | Available    |

Reviewer: we should document v6 as deprecated

###### Gateway URL Params

| Field     | Type    | Description                                   | Accepted Values                                                            |
|-----------|---------|-----------------------------------------------|----------------------------------------------------------------------------|
| v         | integer | Gateway Version to use                        | see [Gateway versions](#DOCS_TOPICS_GATEWAY/gateways-gateway-versions) |
| encoding  | string  | The encoding of received gateway packets      | 'json' or 'etf'                                                            |

> | compress? | string  | The (optional) compression of gateway packets | 'zlib-stream'                                                              |

Reviewer 2: fix table spacing

Author: My autoformatter is really messing me up. I'm just gonna turn it off for now and do a separate pass in the future for spacing.

| [Resumed](#DOCS_TOPICS_GATEWAY/resumed)                                             | response to [Resume](#DOCS_TOPICS_GATEWAY/resume)                                                                                |
| [Reconnect](#DOCS_TOPICS_GATEWAY/reconnect)                                         | server is going away, client should reconnect to gateway and resume                                                              |
| [Invalid Session](#DOCS_TOPICS_GATEWAY/invalid-session)                             | failure response to [Identify](#DOCS_TOPICS_GATEWAY/identify) or [Resume](#DOCS_TOPICS_GATEWAY/resume) or invalid active session |
| [Channel Create](#DOCS_TOPICS_GATEWAY/channel-create)                               | new guild channel created                                                                                                       |
| [Channel Update](#DOCS_TOPICS_GATEWAY/channel-update)                               | channel was updated                                                                                                              |

> | [Channel Delete](#DOCS_TOPICS_GATEWAY/channel-delete)                               | channel was deleted                                                                                                              |

Reviewer 3: Shouldn't it specify guild channel here as well?

Reviewer 2: Does it not fire when DMs are closed?

Reviewer 3: doesnt make much sense to have delete/close when we dont have channel create for dms either

Reviewer 4: Get Channel still exists.

#### Channel Create

> Sent when a new channel is created, relevant to the current user. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object. Does not fire for DMs.

Reviewer 2: maybe
```suggestion
Sent when a new guild channel is created, relevant to the current user. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object.
```

| members     | array of [guild member](#DOCS_RESOURCES_GUILD/guild-member-object) objects | set of guild members                                                                        |
| chunk_index | integer                                                                    | the chunk index in the expected chunks for this response (0 <= chunk\_index < chunk\_count) |
| chunk_count | integer                                                                    | the total number of expected chunks for this response                                       |
| not_found?  | array                                                                      | if passing an invalid id to `REQUEST_GUILD_MEMBERS`, it will be returned here               |
| presences?  | array of [presence](#DOCS_TOPICS_GATEWAY/presence) objects                 | if passing true to `REQUEST_GUILD_MEMBERS`, presences of the returned members will be here  |
| nonce?      | string                                                                     | the nonce used in the [Guild Members Request](#DOCS_TOPICS_GATEWAY/request-guild-members)   |

> 

Reviewer 2: uh, looks like you commented on the Guild Members Chunk area, which is the response to Request Guild Members, which only needs the presence intent to return presences. If you were talking about Guild Create, that's [already documented](https://discord.com/developers/docs/topics/gateway#guild-create) in the right place.
